biogeophys/FatesPlantHydraulicsMod.F90

```diff
@@ -1483,12 +1545,26 @@ subroutine UpdateH2OVeg(nsites,sites,bc_out)
           do while(associated(currentCohort))
              ccohort_hydr => currentCohort%co_hydr
              !only account for the water for not newly recruit for mass balance
-             if(.not.ccohort_hydr%is_newly_recruited) then 
+             if(.not.ccohort_hydr%is_newly_recruited) then
+               ! check for nan value , Junyan
+               do ily = 1,csite_hydr%nlevrhiz
+                  if(ccohort_hydr%th_aroot(ily)/=ccohort_hydr%th_aroot(ily)) then
+                    ccohort_hydr%th_aroot(ily) = 0         
+                  endif                 
+               end do  ! end checking nan
+
+
                 csite_hydr%h2oveg = csite_hydr%h2oveg + &
                      (sum(ccohort_hydr%th_ag(:)*ccohort_hydr%v_ag(:)) + &
                      ccohort_hydr%th_troot*ccohort_hydr%v_troot + &
                      sum(ccohort_hydr%th_aroot(:)*ccohort_hydr%v_aroot_layer(:)))* &
                      denh2o*currentCohort%n
+                if (JD_debug) then 
```

Comment:
Do you need this still?

Reply:
The root model hasn't be extensively tested. I think to keep this for a little while in case something goes wrong.